upload_log_files.py

```diff
@@ -0,0 +1,125 @@
+            continue
+        log.info(f"Deleting {file_path}")
+        os.remove(os.path.join(dir_path, file_path))
+
+# TODO: Move to google_cloud_utils once the upload strategy is has been reviewed
+def list_blobs(bucket_url, prefix, bucket_credentials_file_path):
+
+    storage_client = storage.Client.from_service_account_json(bucket_credentials_file_path)
+    parsed_bucket_url = urlparse(bucket_url)
+    bucket_name = parsed_bucket_url.netloc
+    blobs = storage_client.list_blobs(bucket_name, prefix=prefix)
+    blob_names = [blob.name for blob in blobs]
+
+    return blob_names
+
+def get_uploaded_log_dates(uploaded_log_list, date_pattern):
+
+    dates_match = [re.search(date_pattern, file) for file in uploaded_log_list]
+    uploaded_log_dates = []
+    for date_match in dates_match:
+        if date_match == None:
+            continue
+        uploaded_log_dates.append(date_match.group())
+
+    return uploaded_log_dates
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="Uploads pipeline log files to g-cloud")
+
+    parser.add_argument("user", help="User launching this program")
+    parser.add_argument("google_cloud_credentials_file_path", metavar="google-cloud-credentials-file-path",
+                        help="Path to a Google Cloud service account credentials file to use to access the "
+                             "credentials bucket")
+    parser.add_argument("pipeline_configuration_file_path", metavar="pipeline-configuration-file-path",
+                        help="Path to the pipeline configuration json file")
+    parser.add_argument("memory_profile_dir_path", metavar="memory-profile-dir-path",
+                        help="Path to the memory profile log directory with file to upload")
+    parser.add_argument("data_archive_dir_path", metavar="data-archive-dir-path",
+                        help="Path to the data archive directory with file to upload")
+
+    args = parser.parse_args()
+
+    user = args.user
+    google_cloud_credentials_file_path = args.google_cloud_credentials_file_path
+    pipeline_configuration_file_path = args.pipeline_configuration_file_path
+    memory_profile_dir_path = args.memory_profile_dir_path
+    data_archive_dir_path = args.data_archive_dir_path
+
+    date_pattern = r'\d{4}-\d{2}-\d{2}'
+
+    log.info("Loading Pipeline Configuration File...")
+    with open(pipeline_configuration_file_path) as f:
+        pipeline_configuration = PipelineConfiguration.from_configuration_file(f)
+    Logger.set_project_name(pipeline_configuration.pipeline_name)
+    log.debug(f"Pipeline name is {pipeline_configuration.pipeline_name}")
+
+    log.warning(f"Deleting old memory profile & data archives files from local disk...")
+    delete_old_log_files(memory_profile_dir_path, fetch_latest_modified_file_path(memory_profile_dir_path))
+    delete_old_log_files(data_archive_dir_path, fetch_latest_modified_file_path(data_archive_dir_path))
```

Comment:
Sorry to go round on this again - it's quite hard thinking through all the different ways of sequencing this vs. the pipeline.

If you remove all the old log files, you'll remove previous days that may have failed to upload e.g. because this script stopped. I think the aim here is to upload all previous days that don't have a successful upload only, so I think we want to keep:
 - Recent archives for this pipeline
 - One non-recent archive for each day where we haven't been able to successfully upload yet

and remove everything else

Reply:
Thanks for your patience and keenness in reviewing this. have updated to;

- delete local files if file_date is already in the bucket, else: check for the latest file for that date and delete the rest.
- upload the lates-modified files left in the directory.
- moved list_blobs to pipeline infrastructure.